### PR TITLE
feat: integrate UI with back end

### LINKED ISSUE #8: Show code on Error
Deployment should work only when compilation is successfull

### LINKED ISSUE #10: Integrate front-end with Ai Agents
- Steps
- Toata mamaliga

### PR DESCRIPTION
**Pull request by doinel1a**

### CODE REVIEW

**Comment by doinel1a** [comment]
Implement Generate, Compile & Audit Smart Contract with creations steps, using the back-end.

File: front-end/src/reducers/compile-contract.ts
Comment (urataps) [inline front-end/src/reducers/compile-contract.ts:1]: Here should we have error message beside the artifact?

File: front-end/src/reducers/compile-contract.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import EReducerState from '@/constants/reducer-state';
+
+const compileContractInitialState = {
+  isLoading: false,
+  isError: false,
+  isSuccess: false,
+  artifact: null as string | null
+};
```
Comment (urataps) [inline front-end/src/reducers/compile-contract.ts:8]: Here should we have error message beside the artifact?

**Review by urataps — APPROVED** [review]

File: front-end/src/reducers/compile-contract.ts (hunk 1 above)
Comment (doinel1a) [reply front-end/src/reducers/compile-contract.ts:8]: I guess not... Sometimes it can get up to 500kb / 1mb of error message...

**Pull request merged**